### ISSUE TITLE
App exits to device home screen on playing video from IMDB site

### ISSUE THREAD

**Issue by Sdaswani** [mozilla-mobile/firefox-tv#1412]
Affected devices      : Fire TV 2014 , Fire TV 2015   ,Amazon Fire TV Edition  ,Fire TV Stick with Alexa Voice Remote, Fire TV Cube ,Fire TV (Gen 3),Fire TV Stick.    

Issue Present in Live : Yes

Repro rate            : 5/5

Steps to reproduce:

1. Install and launch the app.

2. Select the IMBD thumbnail from homescreen of the app

3. Once the IMDB site open try to play any video in the site

4. Observe

Actaul Result   : App exits to device home screen playing any videos in IMDB Site.

Expected Result : Video shoud play on selecting .

**Comment by Sdaswani** [comment]
@dnarcese could not reproduce on Fire TV Pendant with Fire OS 6.2.5.3. @psymoon checking with the newer OS 6.2.5.5 .

**Comment by psymoon** [comment]
@Sdaswani Not reproducible with 6.2.5.5

**Comment by liuche** [comment]
Was able to repro this on the Cube, but it looks like a webview crash. Attaching a log here (this is with Turbo mode off):

```
2018-10-30 14:47:13.881 4159-7038/? D/Vizzini:MediaSessionLib.MediaSessionControllerStateMgr: Found valid active controller: org.mozilla.tv.firefox with state: 2
2018-10-30 14:47:13.906 3600-5724/? E/SELinux: avc:  denied  { log_internal_metric } for  scontext=u:r:untrusted_app:s0:c512,c768 tcontext=u:r:system_server:s0 tclass=amazon_policies permissive=0
2018-10-30 14:47:13.908 3600-5724/? D/ActivityManager: cleanUpApplicationRecord -- 7858
2018-10-30 14:47:13.928 3287-3350/? I/oemcrypto_tz: [OEMCrypto_FreeShm:173] OEMCrypto_FreeShm
2018-10-30 14:47:13.937 3287-3350/? I/oemcrypto_tz: [OEMCrypto_FreeShm:173] OEMCrypto_FreeShm
2018-10-30 14:47:13.940 8842-8908/? E/cr_MediaResource: Error configuring data source: java.lang.RuntimeException: setDataSource failed: status = 0xFFFFFFEA
    java.lang.RuntimeException: setDataSource failed: status = 0xFFFFFFEA
        at android.media.MediaMetadataRetriever._setDataSource(Native Method)
        at android.media.MediaMetadataRetriever.setDataSource(MediaMetadataRetriever.java:97)
        at org.chromium.content.browser.MediaResourceGetter.configure(MediaResourceGetter.java:5406)
        at org.chromium.content.browser.MediaResourceGetter.extractMediaMetadata(MediaResourceGetter.java:1139)
2018-10-30 14:47:13.940 8842-8908/? E/cr_MediaResource: Unable to configure metadata extractor
2018-10-30 14:47:13.989 8842-8842/? I/AMPEXO1Factory: createMediaPlayer : 3.0.0 AMPEXO1IMPL Version : 4.0.2 (for Exo 1.5.9) with Config = ExoPlayerSetupConfig{mMinBufferMs =3000, mMinRebufferMs =5000, mMaxDroppedFrameCountToNotify =15, mEnableCrossProtocolRedirection =false, mEnableTunnelModePlayback =false, mMaxRetriesOnConnectionError =2, mEnableEpochTimeInLive =false, mUsePsipCsdForCaptions =false}
2018-10-30 14:47:13.998 8842-8842/? I/StateMachineImplBase: Instantiating and initializing state machine. setup = ExoPlayerSetupConfig{mMinBufferMs =3000, mMinRebufferMs =5000, mMaxDroppedFrameCountToNotify =15, mEnableCrossProtocolRedirection =false, mEnableTunnelModePlayback =false, mMaxRetriesOnConnectionError =2, mEnableEpochTimeInLive =false, mUsePsipCsdForCaptions =false}
2018-10-30 14:47:14.003 8842-8842/? I/ExoPlayerImpl: Init 1.5.9
2018-10-30 14:47:14.006 8842-9488/? I/StateMachineImplBase: Initial state enter
2018-10-30 14:47:14.006 8842-8842/? I/StateMachineImplBase: addStateChangeListener API call
2018-10-30 14:47:14.006 8842-8842/? I/StateMachineImplBase: addInfoListener API call
2018-10-30 14:47:14.006 8842-8842/? I/StateMachineImplBase: addErrorListener API call
2018-10-30 14:47:14.007 8842-8842/? I/StateMachineImplBase: addId3MetadataListener API call
2018-10-30 14:47:14.007 8842-8842/? W/StateMachineImplBase: Subtitle view is not initialized yet. Setting caption style postponed
2018-10-30 14:47:14.007 8842-8842/? I/StateMachineImplBase: setVolume API call= 1.0
2018-10-30 14:47:14.013 8842-9488/? I/StateMachineImplBase: dropping change change callback as prev and current state same. state= IDLE
2018-10-30 14:47:14.021 8842-8842/? I/StateMachineImplBase: setVolume API call= 0.9
2018-10-30 14:47:14.053 8877-8910/? I/chromium: [INFO:HTMLMediaElement.cpp(2505)] Pause video at:0
2018-10-30 14:47:14.055 8877-8910/? I/chromium: [INFO:HTMLMediaElement.cpp(2505)] Pause video at:0
2018-10-30 14:47:14.059 8842-9488/? I/StateMachineImplBase: Message handled in initial state. Message what= unknown_msg_what_id=125 msg: { when=-5ms what=125 obj=1.0 target=com.amazon.mediaplayer.StateMachineUtil.StateMachine$SmHandler }
2018-10-30 14:47:14.062 8842-9488/? I/StateMachineImplBase: Message handled in initial state. Message what= unknown_msg_what_id=125 msg: { when=-40ms what=125 obj=0.9 target=com.amazon.mediaplayer.StateMachineUtil.StateMachine$SmHandler }
2018-10-30 14:47:14.070 8842-8842/? I/StateMachineImplBase: setUserAgent API call
2018-10-30 14:47:14.071 8842-8842/? I/StateMachineImplBase: Open API call. parameters: ContentParameters{url='suppressedcustomSource ='org.chromium.amazon_webview.browser.media.AmazonSampleSource@5de6d58', mimeType=CONTENT_TYPE_CUSTOM_SAMPLE_SOURCE, laurl='suppressed', oobTextSources=null, encryptionSchema=ENCRYPTION_DEFAULT}
2018-10-30 14:47:14.073 8842-9488/? I/ExoPlayerStateMachineImpl: MimeType= CONTENT_TYPE_CUSTOM_SAMPLE_SOURCE, builder= CustomSampleSourceRendererBuilder
2018-10-30 14:47:14.078 8842-9488/? I/AmazonQuirks: ro.amzn.media.tunnelmode: <true>
2018-10-30 14:47:14.078 8842-9488/? I/AmazonQuirks: enableTunnelModeIfSupported: enabled = false
2018-10-30 14:47:14.079 8842-9488/? I/CustomSampleSourceRendererBuilder: buffer config to use= ExoNonAdaptivePlaybackBufferConfig{mBufferSegmentSzBytes=131072, mNumBufferSegments=290, mMaxSamplesForBandwidthMeter=2000, mNumBufferSegmentPreAllocate= 306, mIgnoreDwellModeInMP4= false, mUseMovieHeaderForDurationInMP4= false}
2018-10-30 14:47:14.080 8842-9488/? I/AmazonQuirks: isTunnelModeEnabled? supported = true , enabled = false
2018-10-30 14:47:14.093 8842-9488/? I/AudioTrack: isLatencyQuirkEnabled = false
2018-10-30 14:47:14.093 8842-9488/? I/AudioTrack: isDolbyPassthroughQuirkEnabled = false
2018-10-30 14:47:14.125 8842-9488/? I/ExoPlayerStateMachineImpl: onRenders
2018-10-30 14:47:14.125 8842-9488/? I/StateMachineImplBase: Opened state enter
2018-10-30 14:47:14.136 3600-3621/? I/DisplayManagerService: Display device changed: DisplayDeviceInfo{"Built-in Screen": uniqueId="local:0", 3840 x 2160, modeId 5, defaultModeId 1, supportedModes
...
2018-10-30 14:47:14.194 8842-8842/? I/StateMachineImplBase: setUserAgent API call
2018-10-30 14:47:14.194 8842-8842/? I/StateMachineImplBase: Open API call. parameters: ContentParameters{url='suppressedcustomSource ='org.chromium.amazon_webview.browser.media.AmazonSampleSource@5de6d58', mimeType=CONTENT_TYPE_CUSTOM_SAMPLE_SOURCE, laurl='suppressed', oobTextSources=null, encryptionSchema=ENCRYPTION_DEFAULT}
2018-10-30 14:47:14.198 8842-8842/? I/AmazonMediaPlayerBridge: setSurfaceView called
2018-10-30 14:47:14.198 8842-8842/? I/StateMachineImplBase: setVideoSurfaceContainerView api call: videoSurfaceContainerView= null UI thread= true
2018-10-30 14:47:14.199 8842-8842/? I/StateMachineImplBase: setVideoSurfaceContainerView api call: videoSurfaceContainerView= org.chromium.components.external_video_surface.ExternalVideoSurfaceContainer$NoPunchingSurfaceView{f158eb1 V.E...... ......ID 0,0-1920,1080} UI thread= true
2018-10-30 14:47:14.199 8842-8842/? I/StateMachineImplBase: setSurfaceAndVideoView: videoview parent= android.webkit.WebView{b8ddf96 VFEDHVC.. .F....ID 0,0-1920,1080}
2018-10-30 14:47:14.199 8842-8842/? I/StateMachineImplBase: maybeAddSubtitleViewAndLayout processing 
2018-10-30 14:47:14.201 8842-8842/? I/StateMachineImplBase: setSurface API call. Block= true
2018-10-30 14:47:14.201 8842-9488/? I/StateMachineImplBase: pushing Surface = Surface(name=null)/@0x98c1717blocking = true
2018-10-30 14:47:14.202 8842-9488/? I/StateMachineImplBase: Surface pushed. Notifying.
2018-10-30 14:47:14.221 8842-8842/? I/AmazonMediaPlayerBridge: onPlayerStateChange: OPENING
2018-10-30 14:47:14.226 3780-4115/? V/Avrcp: MediaController playback changed: PlaybackState {state=2, position=0, buffered position=0, speed=0.0, updated=137138748, actions=822, custom actions=[], active item id=-1, error=null}
2018-10-30 14:47:14.226 3780-4115/? V/Avrcp: updatePlaybackState (1): old=PlaybackState {state=2, position=0, buffered position=0, speed=0.0, updated=137132720, actions=822, custom actions=[], active item id=-1, error=null}(2), new=PlaybackState {state=2, position=0, buffered position=0, speed=0.0, updated=137138748, actions=822, custom actions=[], active item id=-1, error=null}(2)
...
2018-10-30 14:47:14.267 8842-8842/? I/StateMachineImplBase: prepare API call
2018-10-30 14:47:14.267 8842-8842/? I/chromium: [INFO:CONSOLE(0)] "XMLHttpRequest cannot load https://v.amazon-adsystem.com/va/fw/getAds?caid=vi713734681&csid=imdb.us.web.mobile;site=imdb.com&marketplace=1&sessionId=01012dab187a2a80073783658dba9a6a89fe8b0accc6716070c5e1e64a04942c715a&clientId=imdb_web_mobile&g=sh&g=n&tt=tv&coo=us&id=vi713734681&id=tt9122526&ab=d&bpx=1&vi=2&vl=ls025720609&vt=player&u=5374146294938125;slid=imdbWebPrerollSlot&ad-sid=01012dab187a2a80073783658dba9a6a89fe8b0accc6716070c5e1e64a04942c715a&api=%5B0%2C%201%2C%202%5D&pj=%7B%22CustomMacroContext%22%3A%7B%22PriorityVideoSource%22%3A%22true%22%7D%7D&vsz=960x540. No 'Access-Control-Allow-Origin' header is present on the requested resource. Origin 'https://m.imdb.com' is therefore not allowed access. The response had HTTP status code 404.", source:  (0)
2018-10-30 14:47:14.267 8842-9488/? I/StateMachineImplBase: pushing Surface = Surface(name=null)/@0x98c1717blocking = false
2018-10-30 14:47:14.268 8842-9488/? I/ExoPlayerStateMachineImpl: onExoPlayerStateChanged: playWhenReady = false state= STATE_PREPARING currentPlayerState (state machine) = Preparing
2018-10-30 14:47:14.274 8842-8842/? I/AmazonMediaPlayerBridge: onPlayerStateChange: PREPARING
2018-10-30 14:47:14.274 8842-8842/? I/ExoPlayerStateMachineImpl: setSelectedTrack api call. type= SUBTITLE_TEXT index= -1
2018-10-30 14:47:14.275 8842-8842/? I/StateMachineImplBase: clearing Cues
2018-10-30 14:47:14.276 8842-8842/? I/ExoPlayerStateMachineImpl: setSelectedTrack api call. type= CLOSED_CAPTION_TEXT index= -1
2018-10-30 14:47:14.276 8842-8842/? I/StateMachineImplBase: clearing Cues
    
    --------- beginning of crash
2018-10-30 14:47:14.511 9492-8910/? A/google-breakpad: -----BEGIN BREAKPAD MICRODUMP-----
2018-10-30 14:47:14.511 9492-8910/? A/google-breakpad: V AndroidWebView:59.0.3071.125
2018-10-30 14:47:14.511 9492-8910/? A/google-breakpad: O A arm 04 armv8l Amazon/stark:6.0/NS6255/1616/N:user/amz-p,release-keys
2018-10-30 14:47:14.511 9492-8910/? A/google-breakpad: P renderer
2018-10-30 14:47:14.512 9492-8910/? A/google-breakpad: G UNKNOWN
2018-10-30 14:47:14.549 9492-8910/? A/google-breakpad: H 12C00000 FFFF1000 0097 32A57000 9A02D000 0C:0A 0D:05 0E:02 0F:03 10:02 11:03 12:02 13:1C 14:0C 15:12 16:18 17:18 18:0E 1B:01 1C:02 1D:01
2018-10-30 14:47:14.568 9492-8910/? A/google-breakpad: S 0 E8F46EBC E8F46000 00003000
2018-10-30 14:47:14.568 9492-8910/? A/google-breakpad: S E8F46000 00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000
...
```

Snipping out some irrelevant looking display state logging.

**Comment by psymoon** [comment]
I can't seem to reproduce this on cube (OS 6.2.5.5). I tried playing 5 videos with turbo on and off. @npark-mozilla can also confirm

**Comment by levisacal** [comment]
Was only able to reproduce 2 times after around 20 tries on Fire TV with 4k and Alexa Voice Remote (OS 6.2.5.5). No specific STR just tried to watch a video on IMBD. Firefox version 3.0.2

**Comment by Sdaswani** [comment]
@Sdaswani to have @haldemann99 have their QA team try to repro on the LAT.

**Comment by Sdaswani** [comment]
@haldeman99 said: "Confirmed with the AWV  team that this requires a fix from AWV. Your next major release  shouldn’t be held up by this. If you have a way to catch the exception  and fail gracefully (i.e. not crash), that would be  good." is catching the exception possible @psymoon ?

**Comment by Baron-Severin** [comment]
I've been unable to reproduce this on 4k, Cube, or Stick.  Could anyone verify that this is still an issue?  It's possible that this has already been addressed on Amazon's end.

**Comment by Sdaswani** [comment]
I can ask Amazon if they can still repro, let me know if i should.

**Comment by Baron-Severin** [comment]
@Sdaswani that would be super helpful, thanks!

**Comment by Sdaswani** [comment]
@Baron-Severin it may also have been fixed by IMDB? I would say if you can't repro per the STRs, go ahead and close. We can also re-open.
